### PR TITLE
Improved setup docs

### PR DESCRIPTION
**Pull request by Caeden117**
Starting this PR a little early. Documentation may not be as fleshed out as whats needed for proper deployment, hopefully some good reviews would point that out.

This PR introduces new setup documentation for Hive, as well as the following:
- Modified `Dockerfile` to pull in the new HiveCorePlugins docker image.
  - This should allow anyone to deploy Hive using Docker / Docker Compose without needing to manually download and set up Hive core plugins.
- Modified `docker-compose.yml` with more comments and examples for volume mounting and plugin configuration.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -17,14 +17,31 @@ services:
...
+      - /path/to/cdn/objects:/app/cdn/objects
+      
+      # Where metadata about CDN objects are stored
+      - /path/to/cdn/metadata:/app/cdn/metadata 
+      
+      # Where Hive permission rules are stored
+      - /path/to/rules:/app/Rules
+      
+      # Where external Hive plugins will be loaded from
+      #- /path/to/hive/plugins:/app/plugins
```
Comment (sc2ad) [inline docker-compose.yml:44]: Commented intentionally?

File: docs/Hive/Setup.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+The Docker Compose file contains some configuration that should be changed before deployment. Please open `docker-compose.yml` in any text environment, and make any necessary modification and configuration to match your deployment environment.
+
+When you are ready to deploy Hive, run `docker-compose up`.
+
+## Downloading Hive
+
+If you do not want to use Docker, you can download the latest release of Hive through [the Releases page](https://github.com/Atlas-Rhythm/Hive/releases).
+
+Simply unzip the Hive release wherever you wish to host Hive.
```
Comment (sc2ad) [inline docs/Hive/Setup.md:21]: Do you also need plugins for this? Or do the releases have plugins bundled with them?

Comment (sc2ad) [reply docs/Hive/Setup.md:21]: If not, might be nice to clarify

**Review by sc2ad — APPROVED** [review]

File: docs/Hive/Setup.md (hunk 2 above)
Comment (Caeden117) [reply docs/Hive/Setup.md:21]: I think ideally, when we do go to make releases, we'd want to bundle HiveCorePlugins into the release zip. A major pain point in deployment right now is having to manually download and install the HiveCorePlugins to even use Hive, so I believe bundling core plugins wherever possible is a good idea.

File: docker-compose.yml (hunk 1 above)
Comment (Caeden117) [reply docker-compose.yml:44]: Correct. HiveCorePlugins is already included in the Hive docker image, so mounting the plugins folder is only necessary if the administrator wants to load third party plugins.

**Pull request merged**